minix/commands/autopart/autopart.c

```diff
@@ -1531,8 +1531,13 @@ void m_read(int ev, int *biosdrive)
 	stat_start(0);
 	fflush(stdout);
 
-	if ((device= open(curdev->name, mode= O_RDWR, 0666)) < 0) {
-		if (device >= 0) { close(device); device= -1; }
+	device = open(curdev->name, O_RDWR, 0666);
+	if (device < 0) {
+		return;
+	}
+	else if (device >= 0) {
+		close(device);
+		device = -1;
```

Comment:
In this case I personally don't mind, but.. as a general rule, it's best not to fix multiple issues in a patch with a single purpose. Even if the unrelated changes improve the code, they make the patch harder to check and might have unintended side effects. For example, here you remove the assignment to the "mode" variable in line 1534, so I had to check whether that variable is used somewhere down the code. It is not, so your patch will now add a new "unused variable" compiler warning, and should actually also remove the declaration of that variable a bit higher up. All this has nothing to do with the original problem, which was just about line 1535, and as such these "ripple effects" complicate the reviewing process, well you get the point :)